### ISSUE TITLE
OpenAPI / SwaggerUI not working with Spring Boot

### ISSUE THREAD

**Issue by goafabric** [hapifhir/hapi-fhir#3698]
Following this documentation 
I added the dependency for the OpenAPI Interceptor
as well as registering it with

registerInterceptor(new OpenApiInterceptor());

While SwaggerUI shows up under /fhir/swagger-ui/
it crashes badly with a lot of errors.
See below for these.
I've also attached my Spring Boot Configuration ...
Tried it with both 5.x + 6.x ..

Am I missing somehthing here ?

[FhirRestfulServerConfiguration.java.zip]
(https://github.com/hapifhir/hapi-fhir/files/8899411/FhirRestfulServerConfiguration.java.zip)

--- cut ---

2022-06-14 13:42:12.715  WARN [fhir-facade,9162f136c2b371a9,9162f136c2b371a9] 27184 --- [io-50700-exec-8] c.u.f.r.s.i.ExceptionHandlingInterceptor : Failure during REST processing: 
ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException: HAPI-0302: Unknown resource type 'swagger-ui' - Server knows how to handle: [Practitioner, Organization, Bundle, Patient, OperationDefinition]

2022-06-14 13:42:12.716  WARN [fhir-facade,15f7728bb5a0720e,15f7728bb5a0720e] 27184 --- [io-50700-exec-7] c.u.f.r.s.i.ExceptionHandlingInterceptor : Failure during REST processing: 
ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException: HAPI-0302: Unknown resource type '*' - Server knows how to handle: [Practitioner, Organization, Bundle, Patient, OperationDe

finition]

2022-06-14 13:42:12.725  WARN [fhir-facade,60a4646158edfdd8,60a4646158edfdd8] 27184 --- [io-50700-exec-9] c.u.f.r.s.i.ExceptionHandlingInterceptor : Failure during REST processing: 
ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException: HAPI-0302: Unknown resource type 'swagger-ui' - Server knows how to handle: [Practitioner, Organization, Bundle, Patient, OperationDefinition]

**Comment by goafabric** [comment]
hello ? anybody in here ?

**Comment by goafabric** [comment]
Yeah thanks for nothing i guess ..

*Problem*
I solved it on my own .. the problem is that the existing Spring Boot Examples are wrong.
And it's a simple "*" that causes the problem.

The examples tell you that you have a serverPath variable somewhere in your application.yaml
that can contain that value
"/fhir/*"

And you will have some kind of configuration classes somewhere that contain:

...
ServletRegistrationBean registration = new ServletRegistrationBean(this, serverPath);
...
setServerAddressStrategy(new HardcodedServerAddressStrategy(serverPath));

While this works without the OpenApi Interceptor, it whoes with OpenAPI because of the "*"

*Solution:*
Your application property should just look like this
"/fhir"

And your implementation something like this:
...
ServletRegistrationBean registration = new ServletRegistrationBean(this, serverPath + "/*");
...
setServerAddressStrategy(new HardcodedServerAddressStrategy(serverPath));

=>

The "*" is needed for the Servlet, but discouraged for HAPI's own serverAddressStrategy


*Value*

The thing is .. the acutal value of all this hassle is not to high.
OpenAPI will now give you the Methos / Operations in charge here.

But the actuall Structure of the FHIR Profiles / DTOs are (currently) missing ..
And this would have been the real value here.
Official HAPI FHIR Test Server behaves the same, but maybe I am missing somehting.
https://hapi.fhir.org/baseR4/swagger-ui/


Closing the issue now on my own
